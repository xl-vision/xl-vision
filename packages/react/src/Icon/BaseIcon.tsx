import React from 'react';
import PropTypes from 'prop-types';
import { env } from '@xl-vision/utils';

export interface BaseIconProps extends React.SVGProps<SVGSVGElement> {
  children: React.ReactElement<React.SVGProps<SVGSVGElement>>;
}

const BaseIcon = React.forwardRef<SVGSVGElement, BaseIconProps>((props, ref) => {
  const { children, ...others } = props;
  return React.cloneElement(children, {
    ...others,
    ref,
  });
});

if (env.isDevelopment) {
  BaseIcon.displayName = 'BaseIcon';

  BaseIcon.propTypes = {
    children: PropTypes.element.isRequired,
  };
}

export default React.memo(BaseIcon);
