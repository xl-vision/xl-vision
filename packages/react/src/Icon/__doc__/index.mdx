# 图标

::: demo ./Basic.tsx

基本用法

基本的使用方式

:::
