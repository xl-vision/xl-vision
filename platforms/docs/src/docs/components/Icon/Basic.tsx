'use client';

import { DeleteFilled, DeleteOutlined } from '@xl-vision/icons';
import { Row } from '@xl-vision/react';

const Basic = () => {
  return (
    <div>
      <Row>
        <Row.Col column={8}>Filled</Row.Col>
        <Row.Col column={16}>
          <DeleteFilled />
        </Row.Col>
      </Row>
      <Row>
        <Row.Col column={8}>Outlined</Row.Col>
        <Row.Col column={16}>
          <DeleteOutlined />
        </Row.Col>
      </Row>
    </div>
  );
};

export default Basic;
