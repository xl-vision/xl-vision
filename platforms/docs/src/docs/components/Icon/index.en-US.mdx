import IconSearch from './IconSearch';

Vector graphics based on [Material Design Icons](https://github.com/google/material-design-icons). Before use icons, you need to install `xl-vision/icons` package:

```bash
npm install --save @xl-vision/icons
```

## Examples

::: demo ./Basic

Basic

Basic

:::

::: demo ./Custom

Custom Icon

Icons can be easily customized via the `createIcon` API provided by the `@xl-vision/icons` package

:::

## Icon Lists

> All the provided `svg` icons are shown here, you can search and view the icons you need

<IconSearch />
