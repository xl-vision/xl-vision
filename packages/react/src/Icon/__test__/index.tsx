import React from 'react';
import { mount } from 'enzyme';
import DeleteFilled from '@xl-vision/icons/DeleteFilled';
import Icon from '..';

describe('Icon', () => {
  it('test render', () => {
    const wrapper = mount(
      <Icon>
        <DeleteFilled />
      </Icon>,
    );

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('test ref', () => {
    const fn = jest.fn();
    mount(
      <Icon ref={fn}>
        <DeleteFilled />
      </Icon>,
    );

    expect((fn.mock.calls[0] as Array<any>)[0]).toBeInstanceOf(SVGSVGElement);
  });
});
