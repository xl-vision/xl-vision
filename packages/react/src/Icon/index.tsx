import React from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import { env } from '@xl-vision/utils';
import { styled } from '../styles';
import BaseIcon, { BaseIconProps } from './BaseIcon';
import ThemeContext from '../ThemeProvider/ThemeContext';

export interface IconProps extends BaseIconProps {}

const IconRoot = styled(BaseIcon, {
  name: 'Icon',
  slot: 'Root',
})(({ theme }) => {
  const { transition } = theme;
  return {
    userSelect: 'none',
    width: '1em',
    height: '1em',
    display: 'inline-block',
    fill: 'currentcolor',
    flexShrink: 0,
    transition: transition.standard('fill', transition.durations.standard),
  };
});

const Icon = React.forwardRef<SVGSVGElement, IconProps>((props, ref) => {
  const { className, ...others } = props;

  const { clsPrefix } = React.useContext(ThemeContext);

  const rootClassName = `${clsPrefix}-icon`;

  const rootClasses = clsx(rootClassName, className);

  return <IconRoot {...others} className={rootClasses} ref={ref} />;
});

if (env.isDevelopment) {
  Icon.displayName = 'Icon';

  Icon.propTypes = {
    viewBox: PropTypes.string,
    className: PropTypes.string,
  };
}

export default Icon;
