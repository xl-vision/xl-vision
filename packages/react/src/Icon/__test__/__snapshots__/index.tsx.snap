// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon test render 1`] = `
.c0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentcolor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  transition: fill 300ms 0ms cubic-bezier(0.4,0,0.2,1);
}

<svg
  class="c0 xl-icon__root"
  viewBox="0 0 24 24"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
  />
</svg>
`;
